### PR TITLE
Fixes #3849 - commands for managing host's interfaces

### PR DESCRIPTION
**Pull request by tstrachota**
- updated structure of `hammer host info`'s output
    - it displays network interfaces now
    - ouptut split to sections similar to UI
- `hammer host {create,update}`
    - accepts nics attributes via `--interface`
    - compute specific attrs in `--interface` are prefixed with
      `compute_`
- new commands `hammer host interface {create,update,delete,info,list}`
- updated documentation

This PR takes advantage of changes introduced in https://github.com/theforeman/hammer-cli/pull/167 and also depends on important API fix https://github.com/theforeman/foreman/pull/2270

### CODE REVIEW

**Comment by mbacovsky** [comment]
[test]

**Comment by mbacovsky** [comment]
[test]

**Comment by mbacovsky** [comment]
[test]

**Comment by tstrachota** [comment]
[test]

**Comment by domcleal** [comment]
Just a reminder, I think this PR and a release is important for Foreman 1.8.0-RC3 (https://groups.google.com/forum/#!topic/foreman-dev/EiFcsDWxzOM) as I don't think the current state of Hammer 0.1.3 and Foreman 1.8 is fully compatible.  Thanks!

**Comment by tstrachota** [comment]
@mbacovsky tests green, can you please re-review?

File: doc/host_create.md [hunk 1]
```diff
@@ -115,9 +163,9 @@ start         # boolean, whether to start the machine or not
 
 Available keys for `--interface`:
 ```
-type              # one of [:bridge, :network]
-network / :bridge # name of interface according to type
-model             # one of [virtio, rtl8139, ne2k_pci, pcnet, e1000]
+compute_type                      # one of [:bridge, :network]
+compute_network / :compute_bridge # name of interface according to type
```
Comment (mbacovsky) [inline doc/host_create.md]: extra colon in `:compute_bridge` ?

File: lib/hammer_cli_foreman/interface.rb [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+        if option_primary == 'true'
+          reset_flag(host_params['host']['interfaces_attributes'], 'primary')
+        end
+        if option_provision == 'true'
+          reset_flag(host_params['host']['interfaces_attributes'], 'provision')
+        end
+        host_params['host']['interfaces_attributes'] += [interface]
+
+        require 'pry'; binding.pry
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/interface.rb]: Debugging leftover...

**Comment by mbacovsky** [comment]
I was trying to swap primary interfaces, but am not sure if that is supported.
I was trying to set  `eth1` interface as --primary end it failed almost as expected:
```
$ hm host interface update --host-id 1 --id 2 --primary 1
Could not update the interface:
  Primary host already has primary interface
  Name has already been taken
```
`Primary host` looks suspicious as wel as the second line.

Turning off primary interface ended up with `500`
```
$ hm host interface update --host-id 1 --id 1 --primary 0
Could not update the interface:
  Host::Base#domain_id= delegated to primary_interface.domain_id=, but primary_interface is nil: #<Host::Managed id: 1, name: "testik.my.lan", last_compile: nil, last_freshcheck: nil, last_report: nil, updated_at: "2015-04-13 14:28:49", source_file_id: nil, created_at: "2014-10-06 15:38:28", root_pass: "$1$R2+ueHfv$45cMU0OttbZahyeKFCEx4/", serial: nil, puppet_status: 0, architecture_id: 1, operatingsystem_id: 4, environment_id: 2, ptable_id: 7, medium_id: 7, build: true, comment: "xxx", disk: "", installed_at: nil, model_id: nil, hostgroup_id: 1, owner_id: 3, owner_type: "User", enabled: true, puppet_ca_proxy_id: nil, managed: true, use_image: nil, image_file: nil, uuid: nil, compute_resource_id: nil, puppet_proxy_id: nil, certname: nil, image_id: nil, organization_id: 1, location_id: 2, type: "Host::Managed", otp: nil, realm_id: nil, compute_profile_id: nil, provision_method: "build", content_source_id: 1, grub_pass: "$1$R2+ueHfv$45cMU0OttbZahyeKFCEx4/", discovery_rule_id: nil, content_view_id: nil, lifecycle_environment_id: nil>
```

Might be my misunderstanding of the concept, but these errors could be more helpful

**Comment by mbacovsky** [comment]
One thing worth considering would be to add attempt to guess host_id from interface_id as adding the redundant host id is annoying.

Besides that it works very well. The host info is a way cleaner now and I like it a lot. I appreciate that the host creation docs were updated as well. Thanks for this PR!

**Comment by mbacovsky** [comment]
I was further trying to switch primary interface with `hm host update --id 1 --interface "name=eth0,primary=false,mac=11:11:11:11:11:11" --interface "name=eth1,primary=true,mac=22:22:22:22:22:22"` which failed with 
```
Could not update the host:
  Mac has already been taken
  Primary host already has primary interface
  Domain can't be blank
  Interfaces.domain can't find domain with this id
  Name has already been taken
```

I tried to drop the two interfaces and re-create them. But dropping primary interface is not allowed. I'm out of ideas here.

**Comment by tstrachota** [comment]
@mbacovsky updated according to your comments.

ad https://github.com/theforeman/hammer-cli-foreman/pull/174#issuecomment-92405041
I turned `--primary` and `--provision` into flags. Switching it of does not make sense since there's always one primary and provision interface required. Weird messages you're observing are API issues. I'm going to file it as separate bugs.

ad https://github.com/theforeman/hammer-cli-foreman/pull/174#issuecomment-92724628
This is not doable now. There's no route that would allow searching for interfaces without providing `host_id` (or `subnet_id` or `domain_id`).

ad https://github.com/theforeman/hammer-cli-foreman/pull/174#issuecomment-92763454
`id` is currently the only unique identifier on NICs. We can't match NIC names as they can be blank. Therefore you need to provide `id` in order to update existing interface. Your example should look like:
```
hm host update --id 1 --interface "id=83,name=eth0,primary=false,mac=11:11:11:11:11:11" --interface "id=32,name=eth1,primary=true,mac=22:22:22:22:22:22"
```
This behaviour is far from ideal but it's the best simplicity vs. usability trade-off I've found so far.

**Comment by tstrachota** [comment]
@mbacovsky one more thing I forgot to mention: Setting `--primary` or `--provision` on some interface automatically removes it from the previous one.

**Comment by mbacovsky** [comment]
Thanks for the improvements @tstrachota. Now it works as I would expect and is more intuitive. Well done!
:+1:  
:swimmer: :ship:
:congratulations:

**Comment by tstrachota** [comment]
Just for the record, issues for the API problems are:
http://projects.theforeman.org/issues/10235
http://projects.theforeman.org/issues/10236

**Pull request merged**